SM60 regression: can't build for Android on Mac

```
rustc 1.29.0-nightly (1ecf6929d 2018-07-16)
info: component 'rust-std' for target 'armv7-linux-androideabi' is up to date
warning: clang: warning: argument unused during compilation: '-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x'
   Compiling mozjs_sys v0.60.0
   Compiling script v0.0.1 (file:///Users/paul/git/servo/components/script)
error: failed to run custom build command for `mozjs_sys v0.60.0`
process didn't exit successfully: `/Users/paul/git/servo/target/debug/build/mozjs_sys-4294d0f258e4cfa8/build-script-build` (exit code: 101)
--- stdout
[[ /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure -ot /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure.in ]] && touch /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure || true
[[ /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure -ot /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure.in ]] && touch /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure || true
! [[ /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure.in -ot /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure ]] && touch /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure || true
! [[ /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure.in -ot /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure ]] && touch /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/old-configure || true
if [[ /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure -nt /Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/config.status ]] ; then \
	  cd /Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out && \
	  PYTHON="" MOZ_TOOLS="" \
	  CC="/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang" CPP="/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -E" CXX="/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++" AR="/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/arm-linux-androideabi/bin/ar" \
	  /Users/paul/.cargo/registry/src/github.com-1ecc6299db9ec823/mozjs_sys-0.60.0/mozjs/js/src/configure --disable-jemalloc --disable-js-shell --disable-tests --build-backends=RecursiveMake --with-arch=armv7-a --with-fpu=neon --with-android-ndk=/Users/paul/git/servo/android-toolchains/ndk --with-android-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 --with-android-version=18 --target=arm-linux-androideabi --disable-gold --disable-shared-js || (cat config.log && exit 1) ; \
	fi
cd /Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out && /Applications/Xcode.app/Contents/Developer/usr/bin/make -f Makefile
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_pre-export
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/_virtualenv/bin/python -m mozbuild.action.process_install_manifest --track install_dist_include.track dist/include _build_manifests/install/dist_include
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/_virtualenv/bin/python -m mozbuild.action.process_install_manifest --track install_dist_public.track dist/public _build_manifests/install/dist_public
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/_virtualenv/bin/python -m mozbuild.action.process_install_manifest --track install_dist_private.track dist/private _build_manifests/install/dist_private
Elapsed: 0.00s; From dist/public: Kept 0 existing; Added/updated 0; Removed 0 files and 0 directories.
Elapsed: 0.00s; From dist/private: Kept 0 existing; Added/updated 0; Removed 0 files and 0 directories.
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/_virtualenv/bin/python -m mozbuild.action.process_install_manifest --track install__tests.track _tests _build_manifests/install/_tests
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/_virtualenv/bin/python -m mozbuild.action.process_install_manifest --track install_dist_bin.track dist/bin _build_manifests/install/dist_bin
Elapsed: 0.05s; From dist/include: Kept 344 existing; Added/updated 0; Removed 0 files and 0 directories.
Elapsed: 0.00s; From dist/bin: Kept 1 existing; Added/updated 0; Removed 0 files and 0 directories.
Elapsed: 0.06s; From _tests: Kept 312 existing; Added/updated 0; Removed 0 files and 0 directories.
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_export
make[4]: Nothing to be done for `host'.
make[4]: Nothing to be done for `export'.
make[4]: Nothing to be done for `export'.
make[4]: Nothing to be done for `export'.
make[4]: Nothing to be done for `export'.
make[4]: Nothing to be done for `export'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_compile
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `host'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
make[4]: Nothing to be done for `target'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_misc
make[3]: Nothing to be done for `recurse_misc'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_libs
make[4]: Nothing to be done for `libs'.
make[4]: Nothing to be done for `libs'.
make[4]: Nothing to be done for `libs'.
make[4]: Nothing to be done for `libs'.
make[4]: Nothing to be done for `libs'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make recurse_tools
make[3]: Nothing to be done for `recurse_tools'.
cargo:rustc-link-search=native=/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src/build
cargo:rustc-link-lib=static=js_static
cargo:rustc-link-search=native=/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/mozglue/build
cargo:rustc-link-lib=static=mozglue
cargo:rustc-link-lib=stdc++
cargo:outdir=/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out
TARGET = Some("armv7-linux-androideabi")
OPT_LEVEL = Some("0")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
TARGET = Some("armv7-linux-androideabi")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
DEBUG = Some("true")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXX_armv7-linux-androideabi = None
CXX_armv7_linux_androideabi = None
TARGET_CXX = None
CXX = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++")
CXXFLAGS_armv7-linux-androideabi = None
CXXFLAGS_armv7_linux_androideabi = None
TARGET_CXXFLAGS = None
CXXFLAGS = Some("--target=armv7-linux-androideabi --sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm --gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64 -I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include -I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include -isystem /Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include -L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x -D__STDC_CONSTANT_MACROS -D__NDK_FPABI__=")
running: "/Users/paul/git/servo/android-toolchains/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "--target=armv7-linux-androideabi" "--sysroot=/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm" "--gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64" "-I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include" "-I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include" "-isystem" "/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include" "-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x" "-D__STDC_CONSTANT_MACROS" "-D__NDK_FPABI__=" "-g" "--target=armv7-linux-androideabi" "-I" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/dist/include" "-I" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src" "-Wall" "-Wextra" "-include" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src/js-confdefs.h" "-DRUST_BINDGEN" "-DSTATIC_JS_API" "-std=gnu++14" "-fno-sized-deallocation" "-Wno-unused-parameter" "-Wno-invalid-offsetof" "-o" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/src/jsglue.o" "-c" "src/jsglue.cpp"
cargo:warning=clang++: warning: argument unused during compilation: '-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x'
cargo:warning=In file included from <built-in>:374:
cargo:warning=In file included from <command line>:5:
cargo:warning=In file included from /Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src/js-confdefs.h:86:
cargo:warning=/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/dist/include/js/RequiredDefines.h:28:9: warning: '__STDC_CONSTANT_MACROS' macro redefined [-Wmacro-redefined]
cargo:warning=#define __STDC_CONSTANT_MACROS
cargo:warning=        ^
cargo:warning=<command line>:1:9: note: previous definition is here
cargo:warning=#define __STDC_CONSTANT_MACROS 1
cargo:warning=        ^
cargo:warning=In file included from src/jsglue.cpp:7:
cargo:warning=src/jsglue.hpp:86:16: warning: private field 'argv_' is not used [-Wunused-private-field]
cargo:warning=    JS::Value* argv_;
cargo:warning=               ^
cargo:warning=src/jsglue.hpp:87:14: warning: private field 'argc_' is not used [-Wunused-private-field]
cargo:warning=    unsigned argc_;
cargo:warning=             ^
cargo:warning=src/jsglue.hpp:88:10: warning: private field 'constructing_' is not used [-Wunused-private-field]
cargo:warning=    bool constructing_:1;
cargo:warning=         ^
cargo:warning=src/jsglue.hpp:89:10: warning: private field 'ignoresReturnValue_' is not used [-Wunused-private-field]
cargo:warning=    bool ignoresReturnValue_:1;
cargo:warning=         ^
cargo:warning=5 warnings generated.
exit code: 0
TARGET = Some("armv7-linux-androideabi")
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
AR_armv7-linux-androideabi = None
AR_armv7_linux_androideabi = None
TARGET_AR = None
AR = Some("/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/arm-linux-androideabi/bin/ar")
running: "/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/arm-linux-androideabi/bin/ar" "crs" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/libjsglue.a" "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/src/jsglue.o"
exit code: 0
TARGET = Some("armv7-linux-androideabi")
cargo:rustc-link-lib=static=jsglue
cargo:rustc-link-search=native=/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out
TARGET = Some("armv7-linux-androideabi")
HOST = Some("x86_64-apple-darwin")
CXXSTDLIB_armv7-linux-androideabi = None
CXXSTDLIB_armv7_linux_androideabi = None
TARGET_CXXSTDLIB = None
CXXSTDLIB = None
TARGET = Some("armv7-linux-androideabi")
cargo:rustc-link-lib=stdc++
Generting bindings ["./src/jsglue.hpp", "--rust-target", "1.19", "--rustified-enum", ".*", "--no-derive-default", "--enable-cxx-namespaces", "--generate", "function,types,vars", "--ignore-methods", "--", "-I", "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/dist/include", "-I", "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src", "-x", "c++", "--target=armv7-linux-androideabi", "--gcc-toolchain=/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64", "-I/Users/paul/git/servo/android-toolchains/ndk/sources/android/support/include", "-I/Users/paul/git/servo/android-toolchains/ndk/sources/cxx-stl/llvm-libc++/libcxx/include", "-isystem", "/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include", "-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x", "-D__STDC_CONSTANT_MACROS", "-D__NDK_FPABI__=", "-DRUST_BINDGEN", "-DSTATIC_JS_API", "-std=gnu++14", "-fno-sized-deallocation", "-Wno-unused-parameter", "-Wno-invalid-offsetof", "-include", "/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/js/src/js-confdefs.h", "--rustfmt-configuration-file", "rustfmt.toml"] clang version 6.0.1 (tags/RELEASE_601/final).

--- stderr
backend.mk:2524: warning: overriding commands for target `../dist/system_wrappers/pixman.h'
backend.mk:1278: warning: ignoring old commands for target `../dist/system_wrappers/pixman.h'
backend.mk:2524: warning: overriding commands for target `../dist/system_wrappers/pixman.h'
backend.mk:1278: warning: ignoring old commands for target `../dist/system_wrappers/pixman.h'
backend.mk:2524: warning: overriding commands for target `../dist/system_wrappers/pixman.h'
backend.mk:1278: warning: ignoring old commands for target `../dist/system_wrappers/pixman.h'
backend.mk:2524: warning: overriding commands for target `../dist/system_wrappers/pixman.h'
backend.mk:1278: warning: ignoring old commands for target `../dist/system_wrappers/pixman.h'
warning: argument unused during compilation: '-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x' [-Wunused-command-line-argument]
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/dist/include/js/RequiredDefines.h:28:9: warning: '__STDC_CONSTANT_MACROS' macro redefined [-Wmacro-redefined]
note: previous definition is here
/usr/include/machine/types.h:37:2: error: architecture not supported
/usr/include/sys/_types/_ptrdiff_t.h:32:9: error: unknown type name '__darwin_ptrdiff_t'
/usr/include/sys/_types/_size_t.h:31:9: error: unknown type name '__darwin_size_t'
/usr/include/sys/_types/_wint_t.h:32:9: error: unknown type name '__darwin_wint_t'
/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include/stdio.h:54:10: fatal error: 'stdarg.h' file not found
warning: argument unused during compilation: '-L/Users/paul/git/servo/android-toolchains/ndk/toolchains/arm-linux-androideabi-4.9/prebuilt/darwin-x86_64/lib/gcc/arm-linux-androideabi/4.9.x' [-Wunused-command-line-argument], err: false
/Users/paul/git/servo/target/armv7-linux-androideabi/debug/build/mozjs_sys-de2e820ce3cdad42/out/dist/include/js/RequiredDefines.h:28:9: warning: '__STDC_CONSTANT_MACROS' macro redefined [-Wmacro-redefined], err: false
/usr/include/machine/types.h:37:2: error: architecture not supported, err: true
/usr/include/sys/_types/_ptrdiff_t.h:32:9: error: unknown type name '__darwin_ptrdiff_t', err: true
/usr/include/sys/_types/_size_t.h:31:9: error: unknown type name '__darwin_size_t', err: true
/usr/include/sys/_types/_wint_t.h:32:9: error: unknown type name '__darwin_wint_t', err: true
/Users/paul/git/servo/android-toolchains/ndk/platforms/android-18/arch-arm/usr/include/stdio.h:54:10: fatal error: 'stdarg.h' file not found, err: true
thread 'main' panicked at 'Should generate JSAPI bindings OK: ()', libcore/result.rs:945:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.

warning: build failed, waiting for other jobs to finish...
```

--- a/etc/ci/buildbot_steps.yml
+++ b/etc/ci/buildbot_steps.yml
@@ -190,8 +190,8 @@ android-mac:
   - ./mach clean-nightlies --keep 3 --force
   - ./mach clean-cargo-cache --keep 3 --force
   - ./etc/ci/bootstrap-android-and-accept-licences.sh
-  - env --unset ANDROID_NDK --unset ANDROID_SDK ./mach build --android --dev
-  - env --unset ANDROID_NDK --unset ANDROID_SDK ./mach package --android --dev
+  - ./mach build --android --dev
+  - ./mach package --android --dev
   - bash ./etc/ci/lockfile_changed.sh
   - ./etc/ci/clean_build_artifacts.sh
 

